libs/wingc/src/ast.rs

```diff
@@ -188,11 +188,15 @@ pub struct FunctionDefinition {
 	/// List of names of function parameters and whether they are reassignable (`var`) or not.
 	pub parameters: Vec<(Symbol, bool)>, // TODO: move into FunctionSignature and make optional
 	/// The function implementation (body).
-	pub statements: Scope,
+	pub statements: Option<Scope>,
 	/// The function signature, including the return type.
 	pub signature: FunctionSignature,
 	/// Whether this function is static or not. In case of a closure, this is always true.
 	pub is_static: bool,
+	/// The `extern` modifier value, pointing to an external implementation file
+	pub external_implementation: Option<String>,
+
```

Comment:
A thought- It's interesting to connect the concept of `extern` and @macro. One could argue that extern is a sort of macro...

We (still) don't support macros in the Wing syntax but perhaps if we did then maybe we could have implemented extern simply using a `require` macro.

```js
@macro("require('./helper.js').uuid()")
inflight uuid(): str;
```

But maybe this is generalizing too quickly...

Reply:
I like this idea, but I don't think we need to go that far yet. I agree with the direction though

Reply:
It's good to have a North Star.